### ISSUE TITLE
Status page for *.decred.org

### ISSUE THREAD

**Issue by peterzen** [decred/dcrweb#1163]
It would be great to set up a status page that monitors the health of:

https://decred.org
https://matrix.decred.org
https://dex.decred.org
... others?

It should monitor 

* port availability
* HTTP return code == 200
* certificate availability (warning when cert about to expire)

**Comment by xaur** [comment]
Also:

- https://dcrdata.decred.org
- https://api.decred.org (for the [VSP API](https://github.com/decred/dcrwebapi))

Maybe the monitoring code could also keep an eye on all the VSP hosts (14 currently).

But this might be quite a bit of work. Is there existing easy to use Go code that you just feed a list of hosts and it does the job?

And should it really be in dcrweb or somewhere else?

**Comment by peterzen** [comment]
First stab at a status page, set up using Uptime Kuma.

Public status page:
https://is.decred.online/status/overview

For internal use:
https://is.decred.online/status/all
https://is.decred.online/status/certificates

Comments, suggestions welcome.

**Comment by peterzen** [comment]
>     * https://dcrdata.decred.org
>     * https://api.decred.org (for the [VSP API](https://github.com/decred/dcrwebapi))
> 
Thanks, added to the PoC.

> 
> Maybe the monitoring code could also keep an eye on all the VSP hosts (14 currently).
> 
Good idea, I'll look into it.

> But this might be quite a bit of work. Is there existing easy to use Go code that you just feed a list of hosts and it does the job?
> 
The PoC is built on Uptime Kuma, that is a decent platform to implement this.

> And should it really be in dcrweb or somewhere else?

It has to be separate from and not dependent in any way on the systems it is monitoring so that when there's an outage it can reliably detect (and send notifications, ideally).

**Comment by xaur** [comment]
Ok thanks. I guess discussing this status page is fine here, since decred.org is the face of everything and when the status page is up we should definitely link to it from decred.org.

**Comment by matthawkins90** [comment]
Amazing. I love Uptime Kuma. This should possibly go into a public repo that people can review/make PRs to.

Sidenote, why can't I visit https://api.decred.org/ even though Uptime Kuma says it's alive? I keep getting 404s in my browser.

**Comment by peterzen** [comment]
> Amazing. I love Uptime Kuma. This should possibly go into a public repo that people can review/make PRs to.
> 
Yeah if this gets adopted then the configuration/customizations would go in a repo.

> Sidenote, why can't I visit https://api.decred.org/ even though Uptime Kuma says it's alive? I keep getting 404s in my browser.

It's a JSON endpoint at https://api.decred.org/?c=vsp

**Comment by xaur** [comment]
@peterzen https://is.decred.online/ is down for me right now
